src/Repository/TufValidatedComposerRepository.php

```diff
@@ -128,191 +51,42 @@ public function __construct(array $repoConfig, IOInterface $io, Config $config,
-                $json = $response->getBody();
-                if ($sha256 !== hash('sha256', $json)) {
-                    if ($retries) {
-                        usleep(100000);
-
-                        continue;
-                    }
-
-                    // TODO use scarier wording once we know for sure it doesn't do false positives anymore
-                    throw new RepositorySecurityException('The contents of '.$filename.' do not match its signature. This could indicate a man-in-the-middle attack or e.g. antivirus software corrupting files. Try running composer again and report this if you think it is a mistake.');
-                }
-
-                $data = $response->decodeJson();
-                HttpDownloader::outputWarnings($this->io, $this->composerRepoUrl, $data);
-
-                if ($cacheKey && !$this->cache->isReadOnly()) {
-                    if ($storeLastModifiedTime) {
-                        $lastModifiedDate = $response->getHeader('last-modified');
-                        if ($lastModifiedDate) {
-                            $data['last-modified'] = $lastModifiedDate;
-                            $json = json_encode($data);
-                        }
-                    }
-                    $this->cache->write($cacheKey, $json);
-                }
-
-                $response->collect();
-
-                break;
-            } catch (\Exception $e) {
-                if ($e instanceof \LogicException) {
-                    throw $e;
-                }
-
-                if ($e instanceof TransportException && $e->getStatusCode() === 404) {
-                    throw $e;
-                }
-
-                if ($retries) {
-                    // Try http if we were trying https.
-                    $this->downgradeToHttp();
-                    if ($this->allowSslDowngrade) {
-                        $filename = str_replace('https://', 'http://', $filename);
-                    }
-
-                    usleep(100000);
-                    continue;
-                }
-
-                if ($e instanceof TufException) {
-                    $e = new RepositorySecurityException('TUF secure error: ' . $e->getMessage(), $e->getCode(), $e);
-                }
-
-                if ($e instanceof RepositorySecurityException) {
-                    throw $e;
-                }
-
-                if ($cacheKey && ($contents = $this->cache->read($cacheKey))) {
-                    if (!$this->degradedMode) {
-                        $this->io->writeError('<warning>' . $this->composerRepoUrl . ' could not be fully loaded (' . $e->getMessage() . '), package information was loaded from the local cache and may be out of date</warning>');
-                    }
-                    $this->degradedMode = true;
-                    $data = JsonFile::parseJson($contents, $this->cache->getRoot() . $cacheKey);
-
-                    break;
-                }
-
-                throw $e;
```

Comment:
Just to document my findings: most of these deletions remove code that was taken straight from the parent method. I don't have enough familiarity with the repo yet to say anything one way or another about the few small differences.